lib/evaluate/check-expression.cc

```diff
@@ -283,7 +283,10 @@ class IsSimplyContiguousHelper
   Result operator()(const CoarrayRef &x) const {
     return CheckSubscripts(x.subscript());
   }
-  Result operator()(const Component &) const { return false; }
+  Result operator()(const Component &x) const {
```

Comment:
Should a scalar component of a scalar base be simply contiguous, if simple scalar are so?

Reply:
I considered that, but it seemed silly to add an extra check for a scalar base when it isn't technically simply contiguous anyway. And then what about the case where there is an array base but the derived type has only one component?

I would be more inclined to follow the standard and treat a simple scalar as not "simply contiguous" if we want to be consistent.

Reply:
So long as things are internally consistent with the treatment of a simple scalar, and a comment to the effect that "the standard defines simple contiguity only as an attribute of arrays, so we arbitrarily return false for scalars" appears, I don't mind.

Reply:
Looking at it further, it seems the simplest way to be consistent is to take your original suggestion and treat a scalar component of a scalar as simply contiguous.